test/test_composite.py

```diff
@@ -10,45 +10,302 @@
+
+        # this keeps updated analysis configuration
+        exp_data1 = par_exp_data.component_experiment_data(1)
+        prob_entry_exp1 = exp_data1.analysis_results("probability")
+        ref_config1 = {"dummyoption": "default_value", "new_config": "test_value"}
+        self.assertDictEqual(prob_entry_exp1.extra["run_options"], ref_config1)
+
+    def test_run_option_overriden(self):
+        """Test if run option is overridden."""
+
+        backend = FakeBogota()
+        exp0 = FakeExperimentCommon(qubits=[0])
+        exp1 = FakeExperimentCommon(qubits=[1])
+
+        # update exp1 run option
+        exp1.set_run_options(shots=2048)
+
+        par_exp = ParallelExperiment([exp0, exp1])
+        par_exp_data = par_exp.run(backend, shots=1024)
+        par_exp_data.block_for_results()
+
+        exp_data1 = par_exp_data.component_experiment_data(1)
+        prob_entry_exp1 = exp_data1.analysis_results("probability")
+        self.assertEqual(prob_entry_exp1.extra["shots"], 1024)
+
+    def test_analysis_hook(self):
+        """Test update class variable with analysis result."""
+
+        class FakeExperimentAnalysisHook(FakeExperimentCommon):
+            """Extract probability and update instance variable."""
+
+            def __init__(self, qubits):
+                super().__init__(qubits)
+                self.probability = None
+
+            def _post_analysis_hook(self, experiment_data):
+                prob_val = experiment_data.analysis_results("probability").value
+                self.probability = prob_val
+
+        backend = FakeBogota()
+        exp0 = FakeExperimentCommon(qubits=[0])
+        exp1 = FakeExperimentAnalysisHook(qubits=[1])
+
+        par_exp = ParallelExperiment([exp0, exp1])
+        par_exp_data = par_exp.run(backend, shots=1024)
+        par_exp_data.block_for_results()
+
+        exp_data1 = par_exp_data.component_experiment_data(1)
+        prob_entry_exp1 = exp_data1.analysis_results("probability")
+
+        self.assertEqual(exp1.probability, prob_entry_exp1.value)
+
+
+class TestBatchExperiment(QiskitTestCase):
+    """Test batch experiment.
+
+    Note:
+        The only difference of this from ``ParallelExperiment`` is the circuit construction.
+        Thus the run and analysis tests can be omitted while keeping the good coverage.
     """
 
-    def test_parallel_options(self):
-        """
-        Test parallel experiments overriding sub-experiment run and transpile options.
-        """
-
-        # These options will all be overridden
-        exp0 = FakeExperiment(0)
-        exp0.set_transpile_options(optimization_level=1)
-        exp2 = FakeExperiment(2)
-        exp2.set_experiment_options(dummyoption="test")
-        exp2.set_run_options(shots=2000)
-        exp2.set_transpile_options(optimization_level=1)
-        exp2.set_analysis_options(dummyoption="test")
-
-        par_exp = ParallelExperiment([exp0, exp2])
-
-        with self.assertWarnsRegex(
```

Comment:
This check should still take place for the run options. Which makes me think, that `CompositeExperiment.set_transpile_options` should raise an error.

Reply:
Good point. You are 100% correct. I added warning instead of rising an error because giving an option doesn't collapse execution though given option doesn't have any effect. See https://github.com/Qiskit/qiskit-experiments/pull/380/commits/2abc6d87ba38e22247a998af40defa4bd7bbc716

Reply:
Thanks. Did you also return `test_parallel_options`?

Reply:
Not exactly. The previous test assumes we cannot apply individual transpile options (this is true for only run options in this PR). So I added separate test that individually evaluate that we cannot apply composite transpile option. Please check this https://github.com/Qiskit/qiskit-experiments/pull/380/commits/4319d70fa550dbfea0e5e75ebbb37a512a2af9f9

Reply:
(I also added reno to explain the difference of behavior)